### PR TITLE
Support elemental HLS manifest style for ads.

### PR DESCRIPTION
**Pull request by masterjk**

### CODE REVIEW

File: reader_test.go [hunk 1]
```diff
@@ -564,6 +566,35 @@ func TestMediaPlaylistWithOATCLSSCTE35Tag(t *testing.T) {
 	}
 }
 
+func TestMediaPlaylistWithEnvivioSCTE35Tag(t *testing.T) {
+	f, err := os.Open("sample-playlists/media-playlist-with-scte35-envivio.m3u8")
+	if err != nil {
+		t.Fatal(err)
+	}
```
Comment (markniebur) [inline reader_test.go:573]: Should there be a `defer os.Close(f)` here?

File: reader_test.go [hunk 2]
```diff
@@ -564,6 +566,35 @@ func TestMediaPlaylistWithOATCLSSCTE35Tag(t *testing.T) {
...
+	p, _, err := DecodeFrom(bufio.NewReader(f), true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	pp := p.(*MediaPlaylist)
+	for i := 0; i < int(pp.Count()); i++ {
+		segment := pp.Segments[i]
+		assert.NotNil(t, segment.SCTE)
+		assert.NotNil(t, segment.SCTE)
```
Comment (markniebur) [inline reader_test.go]: Is checking this value twice intentional?

File: reader_test.go [hunk 3]
```diff
@@ -564,6 +566,35 @@ func TestMediaPlaylistWithOATCLSSCTE35Tag(t *testing.T) {
 	}
 }
 
+func TestMediaPlaylistWithEnvivioSCTE35Tag(t *testing.T) {
+	f, err := os.Open("sample-playlists/media-playlist-with-scte35-envivio.m3u8")
+	if err != nil {
+		t.Fatal(err)
+	}
+	p, _, err := DecodeFrom(bufio.NewReader(f), true)
```
Comment (markniebur) [inline reader_test.go]: Why not skip the cast and have the following:
```
p := NewMediaPlaylist()
p.DecodeFrom(...)
```

File: reader_test.go (hunk 2 above)
Comment (masterjk) [reply reader_test.go]: LOL

File: reader_test.go (hunk 1 above)
Comment (masterjk) [reply reader_test.go:573]: Good catch

File: reader_test.go (hunk 3 above)
Comment (masterjk) [reply reader_test.go]: Changed...

**Pull request merged**